utils/src/main/scala/za/co/absa/enceladus/utils/broadcast/BroadcastUtils.scala

```diff
@@ -42,6 +44,27 @@ object BroadcastUtils {
   def getValueOfSparkExpression(expression: String)(implicit spark: SparkSession): Any = {
     import spark.implicits._
 
+    // This is an not straightforward approach. Let me explain how it works.
+    // The purpose is to convert a Spark expression, like '(0 as code, "n/a" as name) as s' to the actual
+    // value (an instance of Row in case the expression returns a struct).
+    //
+    // Here are examples of Spark expressions and their corresponding values:
+    // "test"                                       -> "test": String
+    // struct("str1" as a, "str2" as b) as s        -> [...]:  Row
+    // array(struct("str1" as a, "str2" as b)) as s -> [...]:  mutable.WrappedArray[Row]]
+    //
+    // In order to convert a Spark expression to a value, a DataFrame is created with a single row and a column
+    // that we will ignore.
+    //
+    // List(1).toDF()       // This creates a DataFrame with a single column 'val' and a single row with a value 1.
+    //
+    // Then we add another column, named 'a' (but the name is not important since we are going to ignore it),
+    // and the value of the Spark expression provided by the user.
+    //
+    // .withColumn("a", expr(expression))
+    //
+    // Then we collect the DataFrame '.collect()', take the first row '(0)' and the second column '(1)', which is
```

Comment:
This last bit is the only thing that I was thinking of and that the `List(1)` and ` column name "a"` are arbitrary and ignored, `expr` is clear enough.

Reply:
After thinking about it, `List(1)` is needed since we need an initial dataframe with 1 row in it. If we create an empty dataset, adding a column with a user-defined expression won't create any row that would contain the actual value of the expression. So will leave `List(1)` and extend the explanation.

The name "a" indeed seems arbitrary, will change it to "dummy".

Reply:
I would actually cut most of the explanation. I just wanted to have a bit explaining the choice of `List(1)`, column name `a` and `collect()(0)(1)`